src/main/io/ledstrip.c

```diff
@@ -257,622 +228,663 @@ void pgResetFn_ledConfigs(ledConfig_t *instance)
+        if(warningFlags & (1 << warningId)) {
+            switch(warningId) {
+                case WARNING_ARMING_DISABLED:
+                    warningColor = colorOn ? &HSV(GREEN)  : &HSV(BLACK);
+                    break;
+                case WARNING_LOW_BATTERY:
+                    warningColor = colorOn ? &HSV(RED)    : &HSV(BLACK);
+                    break;
+                case WARNING_FAILSAFE:
+                    warningColor = colorOn ? &HSV(YELLOW) : &HSV(BLUE);
+                    break;
+                default:;
             }
         }
 
-        for (int ledIndex = 0; ledIndex < ledCount; ledIndex++) {
-            const ledConfig_t *ledConfig = ledConfigs(ledIndex);
-            if (!(ledConfig->flags & LED_FUNCTION_WARNING))
-                continue;
-
-            setLedHsv(ledIndex, warningColor);
-        }
-    }
-
-    if (updateNow && (warningFlags || warningFlashCounter)) {
-        warningFlashCounter++;
-        if (warningFlashCounter == 20) {
-            warningFlashCounter = 0;
-        }
+        applyLedHsv(LED_FUNCTION_WARNING, warningColor);
     }
 }
 
-void applyLedGpsLayer(bool updateNow)
+static void applyLedBatteryLayer(bool updateNow, uint32_t *timer)
 {
-#ifdef CC3D
-    return;
-#endif
-    static uint8_t gpsFlashCounter = 0;
-    static uint8_t gpsPauseCounter = 0;
-    const uint8_t blinkPauseLength = 4;
+    static bool batteryFlash = false;
+    static uint8_t battState;
 
-    const hsvColor_t *gpsColor = colors(specialColors(0)->background);
+    if (updateNow) {
+        battState = getBatteryState();
+        if (battState == BATTERY_OK) {
+            batteryFlash = true;
+            *timer += LED_STRIP_HZ(1);
+        }
+        else {
+            batteryFlash = !batteryFlash;
 
-    if (GPS_numSat == 0 || !sensors(SENSOR_GPS)) {
-        gpsColor = colors(specialColors(0)->gps_nosats);
-        gpsFlashCounter = gpsPauseCounter = 0; // reset counters
-    } else {
-        if (gpsPauseCounter == 0 && (gpsFlashCounter & 1) == 0) {
-            gpsColor = STATE(GPS_FIX) ? colors(specialColors(0)->gps_locked) : colors(specialColors(0)->gps_nolock);
-        } else {
-            gpsColor = STATE(GPS_FIX) ? colors(specialColors(0)->background) : colors(specialColors(0)->gps_nosats);
+            if (battState == BATTERY_WARNING)
+                *timer += LED_STRIP_HZ(2);
+            else
+                *timer += LED_STRIP_HZ(5);
         }
     }
 
     for (int i = 0; i < ledCount; ++i) {
         const ledConfig_t *ledConfig = ledConfigs(i);
+        if (!(ledConfig->function_flags & LED_FLAG_FUNCTION(LED_FUNCTION_BATTERY)))
+            continue;
 
-        if (ledConfig->flags & LED_FUNCTION_GPS) {
-            setLedHsv(i, gpsColor);
+        if (batteryFlash) {
+            hsvColor_t color = HSV(RED);
+            int fuel = calculateBatteryCapacityRemainingPercentage();
+            // set to green when fuel > 80% and red when < 20%
```

Comment:
Is it necessary to strip first and last 20%? What if the angle goes from -20 to 140 deg? But I'am not sure how it looks in practice ...